### ISSUE TITLE
Update `GattRequestRejectedException` with error code from `writeCharacteristic`

### ISSUE THREAD

**Issue by ArtemBurmistrov** [JuulLabs/kable#632]
In Android 13 the [`BluetoothGatt.writeCharacteristic`](https://developer.android.com/reference/kotlin/android/bluetooth/BluetoothGatt#writecharacteristic) function with `boolean` return type was deprecated in favor of [a similar one](https://developer.android.com/reference/kotlin/android/bluetooth/BluetoothGatt#writecharacteristic_1) with `int` return type.
It would be beneficial if `GattRequestRejectedException` could be updated with additional error codes so library users can have more context on why an exception was thrown.

**Comment by twyatt** [comment]
Thanks for pointing this out, and linking to the relevant APIs. 👍 
Seems like a quick/easy addition, I'll try to get to this soon.

**Comment by nooraujla90** [comment]
https://juullabs.atlassian.net/browse/CA-2870

**Comment by twyatt** [comment]
@ArtemBurmistrov, #648 provides the error type via an exception property. Can you try it, and let me know if it fits your needs?:

```kotlin
repositories {
    maven("https://oss.sonatype.org/content/repositories/snapshots")
}

dependencies {
    implementation("com.juul.kable:core:0.29.1-issue-632-1-SNAPSHOT")
}
```

**Comment by ArtemBurmistrov** [comment]
Hey @twyatt, that looks awesome, thank you! I checked PR and it looks like this is exactly what I need.

**Comment by twyatt** [comment]
Shipped in [0.30.0](https://github.com/JuulLabs/kable/releases/tag/0.30.0).
